### DISCUSSION TITLE
Use Visual Queries to access MariaDB

### DISCUSSION THREAD

**Discussion by LIfBi** [2sic/2sxc#2676] [Q&A] [answered]
Just a quick question about the SQLDatasource: Should it be possible to access a MariaDB or MySQL Server to query some data? Right I always get the error "Error getting Stream / reading underlying list".

The Datasource works fine on MS-SQL Connections Strings to access data from a external database, but i don´t get a connection to a mariadb database running. Can I find a better error message, that would give me a better idea why. Or is the Datasource fixed to MS-SQL?

**Comment by iJungleboy** [comment, marked as answer]
Internally it uses `System.Data.SqlClient`

According to my quick research, this is MS Sql only. 

This has not been a requirement so far. You could use the code from that data source and create your own.

**Reply by rockgobo** [reply]
Thanks for that quick answer. I will try to set up a development installation of 2SXC. I am really looking forward to get some insights how such a great project is done from the inside.

**Reply by iJungleboy** [reply]
That would be awesome 🚀

The docs for this aren't as great as we wish they would be, but this checklist should get you started: https://azing.org/2sxc/r/D8cY_BwG
